app/javascript/scripts/components/form_components/providers_autocomplete.js

```diff
@@ -83,14 +83,16 @@ const setProviderHiddenField = (value) => {
   idElement.value = value.id
 }
 
-const suggestionTemplate = (value) => {
-  if (!value) {
-    return
-  }
-
-  const hints = [`Code ${value.code}`, value.name, value.ukprn].filter(Boolean)
-
-  return `${value.name} <span class="autocomplete__option--hint">${hints.join(', ')}</span>`
+const suggestionTemplate = (result) => {
```

Comment:
```suggestion
const suggestionTemplate = (result) => {
  if (!result) return "";

  if (typeof result === "string") return result;

  const hints = [`Code ${result.code}`, result.name, result.ukprn].filter(Boolean);

  return `${result.name} <span class="autocomplete__option--hint">${hints.join(", ")}</span>`;
};
```

Reply:
Prefer not doing return statement in that manner as we be at the mercy of the complex object if it clears the first few hurdles.